src/control/describeIndex.ts

```diff
@@ -9,7 +9,13 @@ export type DescribeIndexOptions = IndexName;
 /** The description of your index returned from { @link Pinecone.describeIndex } */
 export type IndexDescription = IndexMeta;
 
-export const describeIndex = (api: IndexOperationsApi) => {
+export const describeIndex = (
+  api: IndexOperationsApi,
+  callback?: (
```

Comment:
I went back and forth on a few different means of allowing the `describeIndex` call to interact with `HostUrlSingleton`. Initially I wanted to use it within here, but then we didn't have access to the `PineconeConfiguration` for the `cacheKey` in set, so went with a callback.